### PR TITLE
Add tutorial as the second demo.

### PR DESCRIPTION
**Pull request by ryzhyk**
I created a new demo based on the "Feldera: The Basics" tutorial.  The idea is that people who are too lazy to go through all the steps in the tutorial can just play with the resulting pipeline.

- Added missing Python API to create URL input connectors.
- Added `project_demo06-SupplyChainTutorial` demo, which creates and configures the pipeline described in the tutorial using API with one difference:  it doesn't set Kafka broker address in the connectors, so that the pipeline picks them up from the `REDPANDA_BROKERS` env variable. This way the demo works both in docker and locally.
- Modified `docker-compose.yml` to create the new demo, along with the SecOps demo in the demo profile.
  - In the process, added a dependency on the redpanda container to the demo, so it doesn't start until redpanda is up and running, and reduced the demo startup timeout from 5 to 1 seconds.  Perhaps even that is unnecessary.
- Updated the tutorial to point impatient users to the pre-made demo.

One issue with this is that we now compile two programs when running with `--profile demo`, so it takes longer for the demo pipeline to become available.  We could use a separate profile for the tutorial, but that's going to become confusing and unwieldy.

Is this a user-visible change (yes/no): yes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,13 +7,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [0.1.5] - 2023-10-10
+## [0.1.7] - 2023-10-10
```
Comment (lalithsuresh) [inline CHANGELOG.md:10]: This PR should be under [Unreleased]. We haven't released 0.1.7 yet, so don't change what was under 0.1.5.

File: deploy/docker-compose.yml [hunk 2]
```diff
@@ -90,5 +99,13 @@ services:
...
       - bash
       - -c
-      # Run the SecOps demo after giving the service some time to initialize
-      - "sleep 5 && cd demo/project_demo00-SecOps && python3 run.py --dbsp_url http://pipeline-manager:8080 --actions prepare create compile $${SECOPS_DEMO_ARGS}"
+      # Run the demos after giving the service some time to initialize
+      # Start the two demos in parallel.  Compilation will still happen serially, but
+      # this way the user will see the two programs and pipelines straight away.
+      - "sleep 1\ 
+
+        \ (cd demo/project_demo00-SecOps && python3 run.py --dbsp_url http://pipeline-manager:8080 --actions prepare create compile $${SECOPS_DEMO_ARGS})&\
```
Comment (lalithsuresh) [inline deploy/docker-compose.yml:107]: What exactly is the exit code behavior when running both these commands in the background like this? Also, what gets logged to stdout/stderr?

File: CHANGELOG.md (hunk 1 above)
Comment (ryzhyk) [reply CHANGELOG.md:10]: Sorry, brain fart.

File: deploy/docker-compose.yml (hunk 2 above)
Comment (ryzhyk) [reply deploy/docker-compose.yml:107]: They both write to the console, so we'll see logs from both.  The exit status is 0, which .. may not be what we want.  So, what do we want this container to do?  Exit when one of the demos fails or keep running the other one?

Comment (lalithsuresh) [reply deploy/docker-compose.yml:107]: We run integration tests in the Earthfile where we rely on the container's exit code. See for example: https://github.com/feldera/feldera/blob/cec50ad3f81478c3ff2b57fdc4962b9887bde600/Earthfile#L535.

We'd need to make sure the exit codes are reliable in this case. So if even one of the demos fails, it should reflect as such in the container's exit code.

File: demo/project_demo06-SupplyChainTutorial/run.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+def make_config(project):
+    config = DBSPPipelineConfig(
+        project,
+        8,
+        "Feldera Basics Tutorial Pipeline",
+        "See https://www.feldera.com/docs/tutorials/basics/",
+    )
+
+    config.add_url_input(
```
Comment (mihaibudiu) [inline demo/project_demo06-SupplyChainTutorial/run.py:37]: this looks like a good candidate for a helper function

File: docs/tutorials/basics/part3.md [hunk 4]
```diff
@@ -106,6 +106,8 @@ rpk -X brokers=127.0.0.1:19092 cluster metadata
 
 ### Create input/output topics
 
+
```
Comment (mihaibudiu) [inline docs/tutorials/basics/part3.md:109]: do these empty lines have a useful purpose?

File: python/dbsp/connector.py [hunk 5]
```diff
@@ -70,7 +77,8 @@ def save(self):
                 config=ConnectorConfig.from_dict(self.config),
             )
             response = new_connector.sync_detailed(
-                client=self.api_client, json_body=body).unwrap("Failed to create the connector")
+                client=self.api_client, json_body=body
+            ).unwrap("Failed to create the connector")
```
Comment (mihaibudiu) [inline python/dbsp/connector.py:81]: there is unwrap in python too now?

File: python/dbsp/pipeline.py [hunk 6]
```diff
@@ -285,18 +353,27 @@ def status(self) -> Pipeline:
...
-        status = get_pipeline.sync_detailed(client=self.api_client, pipeline_id=self.pipeline_id).unwrap(
-            "Failed to retrieve pipeline status")
+        status = get_pipeline.sync_detailed(
+            client=self.api_client, pipeline_id=self.pipeline_id
+        ).unwrap("Failed to retrieve pipeline status")
         return status
 
-    def wait_for_status(self, expected_status: PipelineStatus, timeout: float = sys.maxsize):
-        start = time.time();
+    def wait_for_status(
```
Comment (mihaibudiu) [inline python/dbsp/pipeline.py:361]: are you doing this reformatting by hand or is there some tool?

**Review by mihaibudiu — APPROVED** [review]

File: deploy/docker-compose.yml (hunk 2 above)
Comment (ryzhyk) [reply deploy/docker-compose.yml:107]: Fair enough

File: python/dbsp/pipeline.py (hunk 6 above)
Comment (ryzhyk) [reply python/dbsp/pipeline.py:361]: I think this is vscode's doing.

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**